packages/blast-core/cmd/blast-cmd.js

```
@@ -9,37 +9,42 @@ const { customTasks } = require('../utilities/task')
 const { getConfig } = require('../utilities/config-utils')
 
 async function main() {
-  if (hideBin(process.argv)[0] !== 'init') {
-    getConfig()
```

Actually, removing the `getConfig` alone is sufficient! Any command which requires `blast.config.js` also calls `getConfig` itself. (`getConfig` doesn't store the config it loads anywhere, it just returns it. So anything which actually wants the config _must_ call it again.)

It looks like `getConfig` is only called _here_ to make blast fail immediately if there isn't a config, but it will always fail later if a command tries to actually use the config.

As it stands, putting `getConfig` inside a try block is actually the same as not doing it, because the _only_ effect `getConfig` has here is to throw an immediately ignored exception if the config isn't there, and do nothing if it is.

---

That is true for everything but the custom task command. Custom tasks were implemented in a different way from all other commands, with no entry points nor config loading (even though it needs it). The getConfig on main was in fact the only way custom tasks were getting the config, and simply transferring it to the task class won't work (at least with no exception handling or refactoring), as it will trigger a "config file now found" exception when out of a project folder.